### PR TITLE
New exercise: Grains

### PR DESCRIPTION
**Pull request by marianfoo**
## Source
https://github.com/exercism/problem-specifications/tree/main/exercises/grains

## Test Exercise
because of the large numbers, only decfloat34 worked of the Predefined Numeric Types  
Even with int8 there are overflows
image

## SAP Unit Tests
image

### CODE REVIEW

**Comment by larshp** [comment]
decfloat34 type missing in transpiler, https://github.com/abaplint/transpiler/issues/588

File: exercises/practice/grains/zcl_grains.clas.abap [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+CLASS zcl_grains IMPLEMENTATION.
+  METHOD square.
+    IF input < 1 OR input > 64.
+      RAISE EXCEPTION NEW cx_parameter_invalid( ).
+    ENDIF.
+    result = 1.
+    DO input - 1 TIMES.
+      result = result * 2.
```
Comment (larshp) [inline exercises/practice/grains/zcl_grains.clas.abap:32]: remove solution from this file?

Comment (marianfoo) [reply exercises/practice/grains/zcl_grains.clas.abap]: true, removed

**Comment by larshp** [comment]
https://github.com/abaplint/transpiler/issues/595

**Comment by mbtools** [comment]
no need for `decfloat34`. solution works fine with type `p`

```abap
    TYPES: type_result TYPE p LENGTH 16 DECIMALS 0.
    METHODS square
      IMPORTING
        !input        TYPE i
      RETURNING
        VALUE(result) TYPE type_result
      RAISING
        cx_parameter_invalid .
    METHODS total
      RETURNING
        VALUE(result) TYPE type_result
      RAISING
        cx_parameter_invalid .
```

**Comment by marianfoo** [comment]
you were right @mbtools thanks
changed it to `p` and it works
Do we have to worry about the abaplint error?

**Comment by mbtools** [comment]
yes.

It's now an abapGit repo, this needs `package.devc.xml` and the `xml` for the class. Best to push from a SAP system using abapGit. This will also make sure the end-of-file for code is proper (i.e. a newline).

Or copy  `package.devc.xml` and a class `xml` from another folder and make the necessary changes.

**Comment by marianfoo** [comment]
@mbtools my dev 1909 was a mess, so the abapGit Setup was a opportunity to set it up proberly
all green now 🥳 (was also finally time)

**Review by mbtools — APPROVED** [review]
LGTM 👍

**Comment by marianfoo** [comment]
exercise is only and green 👌
image

**Comment by mbtools** [comment]
My solution works on website but fails on Docker image:

https://exercism.org/tracks/abap/exercises/grains/solutions/mbtools

image

ABAP seems to have a slight problem with accuracy of type `p`...

**Pull request merged**